anton_gadzikovskiy/graph_generator.cpp

```diff
@@ -65,12 +57,18 @@ void generate_yellow_edges(uni_cource_cpp::Graph& graph) {
           vector<uni_cource_cpp::Graph::VertexId> vertices_on_deeper_depth;
           for (const auto& to_vertex_id :
                graph.get_vertex_ids_on_depth(current_depth + 1)) {
-            if (!graph.is_connected(from_vertex_id, to_vertex_id)) {
+            bool is_connected;
+            {
+              const lock_guard lock(new_edge_mutex);
+              is_connected = graph.is_connected(from_vertex_id, to_vertex_id);
+            }
```

Comment:
Нехорошо оставлять неинициализированные и неконстантные переменные
```suggestion
            const bool is_connected = []() {
              const lock_guard lock(new_edge_mutex);
              return graph.is_connected(from_vertex_id, to_vertex_id);
            }();
```